features/backup/src/main/java/ir/mehdiyari/krypt/backup/DataViewModel.kt

```diff
@@ -57,12 +58,12 @@ internal class DataViewModel @Inject constructor(
     }
 
 
-    fun backupNow() {
+    fun backupNow(uri: Uri) {
         if (backupJob == null) {
             backupJob = viewModelScope.launch(ioDispatcher) {
                 _backupViewState.emit(BackupViewState.Started)
                 try {
-                    if (backupRepository.backupAll()) {
+                    if (backupRepository.backupAll(uri)) {
```

Comment:
Please use `  backupFile.createNewFile()` inside view model, if we do this, there is no need to depend the data layer to the android related classes.